web/src/main/java/com/microsoft/applicationinsights/web/internal/httputils/HttpServerHandler.java

```diff
@@ -121,4 +134,19 @@ public void handleException(Exception e) {
             // swallow AI Exception
         }
     }
+
+    /**
+     * Remove data from Threadlocal and ThreadLocalCleaners
+     */
+    private void cleanup() {
+        try {
+            ThreadContext.remove();
```

Comment:
What would we normally do in these cleaners? Do we anticipate them to access the threadcontext?

Reply:
In the cleaners we clean up the context object set in ThreadLocal. This is called when handleEnd() is triggered and is important.

Reply:
@reyang I have moved the context cleaning logic after cleaners run.